test/parallel.mjs

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env node
+
+import { once } from 'node:events'
+import { spawn } from 'node:child_process'
+import fs from 'node:fs/promises'
+import process from 'node:process'
+import { fileURLToPath } from 'node:url'
+
+const PARALLEL_DIR = new URL('./parallel/', import.meta.url)
+const dir = await fs.opendir(PARALLEL_DIR)
+
+for await (const { name } of dir) {
```

Comment:
this is kind of what `node --test` does but I understand why we relying on it is risky here